test/unit/module/core/reduction_test.hpp

```diff
@@ -0,0 +1,232 @@
+      }
+    }
+  }
+
+  template<typename Callable, typename T, typename C>
+  void call_cx(Callable callable, T v, C cx) const {
+    const auto manual_res = invoke_truth_fn<TruthFn>(v, cx);
+    const auto res = callable[cx](v);
+
+    TTS_ULP_EQUAL(res, manual_res, expected_ulp);
+
+    if constexpr (eve::simd_value<T>)
+    {
+      auto res_splat = callable[eve::splat][cx](v);
+      TTS_ULP_EQUAL(res_splat.get(0), manual_res, expected_ulp);
+
+      for (std::ptrdiff_t i = 0; i < T::size(); ++i)
+      {
+        TTS_EQUAL(res_splat.get(0), res_splat.get(i));
+      }
+    }
+  }
+};
+
+template<typename TruthFn>
+constexpr LogicalTestRunner<TruthFn> logical_runner{};
+
+template<typename TruthFn>
+constexpr ArithmeticTestRunner<TruthFn> arithmetic_runner{};
+
+template<typename TruthFn, typename Callable, typename T, typename Runner>
+void test_case(Callable callable, T v, Runner runner)
+{
+  runner.call_nocx(callable, v);
+  runner.call_cx(callable, v, true);
+  runner.call_cx(callable, v, false);
+  runner.call_cx(callable, v, eve::ignore_none);
+
+  if constexpr (eve::simd_value<T>)
+  {
+    constexpr auto cardinal = eve::cardinal_v<T>;
+
+    using LT = eve::as_logical_t<T>;
+    LT m = tts::poison(LT{ [](auto i, auto) { return i % 2 == 0; } });
+    runner.call_cx(callable, v, m);
+
+    runner.call_cx(callable, v, eve::ignore_all);
+    runner.call_cx(callable, v, eve::keep_first(0));
+
+    if constexpr (cardinal >= 2)
+    {
+      runner.call_cx(callable, v, eve::ignore_extrema(1, 1));
+      runner.call_cx(callable, v, eve::ignore_extrema(cardinal / 2, cardinal / 2));
+      runner.call_cx(callable, v, eve::ignore_extrema(cardinal / 4, cardinal / 4));
+    }
+  }
+}
+
+template<typename TruthFn, typename Callable, typename T>
+void logical_test_case(Callable callable, T v)
+{
+  constexpr auto runner = logical_runner<TruthFn>;
+  test_case<TruthFn>(callable, v, runner);
+}
+
+template<typename TruthFn, typename Callable, typename T>
+void logical_simd_test_cases(Callable callable, eve::as<T> typ)
+{
+  constexpr auto cardinal = eve::cardinal_v<T>;
+  constexpr auto runner = logical_runner<TruthFn>;
+
+  test_case<TruthFn>(callable, eve::true_(typ), runner);
+  test_case<TruthFn>(callable, eve::false_(typ), runner);
+
+  for (std::ptrdiff_t j = 0; j < cardinal; ++j)
+  {
+    eve::logical<T> rhs1 = {}, rhs2 = {}, rhs3 = {}, rhs4 = {};
+
+    for (std::ptrdiff_t i = 0; i < cardinal; ++i)
+    {
+      rhs1.set(i, i >= j ? true : false);
+      rhs2.set(i, i <= j ? false : true);
+      rhs3.set(i, i == j ? true : false);
+      rhs4.set(i, i == j ? false : true);
+    }
+
+    test_case<TruthFn>(callable, rhs1, runner);
+    test_case<TruthFn>(callable, rhs2, runner);
+    test_case<TruthFn>(callable, rhs3, runner);
+    test_case<TruthFn>(callable, rhs4, runner);
+  }
+}
+
+template<typename TruthFn, typename Callable, typename T>
+void arithmetic_test_case(Callable callable, T v, double expected_ulp = 0.0)
```

Comment:
I feel like the word `reduction` is missing somewhere.

Why is just having the word test case enough?